biz.aQute.bndlib/src/aQute/bnd/osgi/Builder.java

```diff
@@ -1294,11 +1298,33 @@ private void extractFromJar(Jar jar, String source, String destination, boolean
 
 			error("Can not find JAR file '%s'", source);
 		} else {
+			Function<String, String> nameMapper = v -> v;
+			if (isTrue(extra.get("flatten:"))) {
+				nameMapper = path -> Strings.getLastSegment(path, '/');
+			} else if (instr != null) {
+				if (extra.containsKey("rebase:")) {
+					/*
+					 * TODO This is broken. The string cannot be both a glob,
```

Comment:
I rebased on master, made some changes and found that the `rebase` directive is broken.  See the comment here.